core/unit_test/TestViewTypedefs.cpp

```diff
@@ -0,0 +1,210 @@
+
+  using anonymous_device_type = Kokkos::Device<typename v_t::execution_space, Kokkos::AnonymousSpace>;
+  static_assert(std::is_same_v<typename v_t::uniform_nomemspace_type,
+                               Kokkos::View<typename v_t::data_type, uniform_layout_type,
+                                            anonymous_device_type, Kokkos::MemoryTraits<0>>>);
+  static_assert(std::is_same_v<typename v_t::uniform_const_nomemspace_type,
+                               Kokkos::View<typename v_t::const_data_type, uniform_layout_type,
+                                            anonymous_device_type, Kokkos::MemoryTraits<0>>>);
+  static_assert(std::is_same_v<typename v_t::uniform_runtime_nomemspace_type,
+                               Kokkos::View<typename data_analysis<T>::runtime_data_type, uniform_layout_type,
+                                            anonymous_device_type, Kokkos::MemoryTraits<0>>>);
+  static_assert(std::is_same_v<typename v_t::uniform_runtime_const_nomemspace_type,
+                               Kokkos::View<typename data_analysis<T>::runtime_const_data_type, uniform_layout_type,
+                                            anonymous_device_type, Kokkos::MemoryTraits<0>>>);
+
+
+  // ==================================
+  // mdspan compatibility
+  // ==================================
+
+  static_assert(std::is_same_v<typename v_t::layout_type, L>);
+  // Not supported yet
+  // static_assert(std::is_same_v<typename v_t::extents_type, >);  
+  // static_assert(std::is_same_v<typename v_t::mapping_type, >);
+  // static_assert(std::is_same_v<typename v_t::accessor_type, >);
+
+  static_assert(std::is_same_v<typename v_t::element_type, ValueType>);
+  // should be remove_const_t<element_type>
+  static_assert(std::is_same_v<typename v_t::value_type, ValueType>);
+  // should be extents_type::index_type
+  static_assert(std::is_same_v<typename v_t::index_type, typename S::memory_space::size_type>);
+  static_assert(std::is_same_v<typename v_t::size_type, std::make_unsigned_t<typename v_t::index_type>>);
+  static_assert(std::is_same_v<typename v_t::rank_type, size_t>);
+
+  // should come from accessor_type
+  static_assert(std::is_same_v<typename v_t::data_handle_type, typename v_t::pointer_type>);
+  static_assert(std::is_same_v<typename v_t::reference, typename v_t::reference_type>);
+  return true;
+};
+
+template<class T, class ... ViewArgs>
+struct ViewParams {};
+
+template<class L, class S, class M, class HostMirrorSpace, class ValueType, class ReferenceType, class T, class ... ViewArgs>
+constexpr bool test_view_typedefs(ViewParams<T, ViewArgs...>) {
+  return test_view_typedefs_impl<Kokkos::View<T, ViewArgs...>, Kokkos::ViewTraits<T, ViewArgs...>,
+                            T, L, S, M, HostMirrorSpace, ValueType, ReferenceType>();
+}
+
+static_assert(test_view_typedefs<
+                  typename Kokkos::DefaultExecutionSpace::array_layout,
```

Comment:
```suggestion
                  Kokkos::DefaultExecutionSpace::array_layout,
```
Same comment everywhere below

Reply:
removed